spec/jobs/plexer_job_spec.rb

```diff
@@ -1,43 +1,88 @@
 require 'rails_helper'
 
 describe PlexerJob, type: :job do
-  let(:job) { described_class.new }
+  let(:job) { described_class.new(druid, version, metadata).tap { |j| j.zip = DruidVersionZip.new(druid, version) } }
   let(:druid) { 'bj102hs9687' }
+  let(:endpoint) { create(:archive_endpoint, delivery_class: 1) } # default
   let(:version) { 1 }
+  let(:md5) { 'd41d8cd98f00b204e9800998ecf8427e' }
+  let(:metadata) do
+    {
+      checksum_md5: md5,
+      size: 123,
+      zip_cmd: 'zip -xyz ...',
+      zip_version: 'Zip 3.0 (July 5th 2008)'
+    }
+  end
+  let(:po) { create(:preserved_object, druid: druid, current_version: version) }
+
+  before { allow(S3EndpointDeliveryJob).to receive(:perform_later).with(any_args) }
+
+  it 'descends from DruidVersionJobBase' do
+    expect(job).to be_an(DruidVersionJobBase)
+  end
 
-  it 'descends from ApplicationJob' do
-    expect(job).to be_an(ApplicationJob)
+  it 'raises without enqueueing if metadata is incomplete' do
+    expect { described_class.perform_later(druid, version, metadata.merge(size: nil)) }
+      .to raise_error(ArgumentError, /size/)
+    expect { described_class.perform_later(druid, version, metadata.reject { |x| x == :zip_cmd }) }
+      .to raise_error(ArgumentError, /zip_cmd/)
   end
 
   describe '#perform' do
-    before { allow(job).to receive(:targets).and_return([S3EndpointDeliveryJob]) }
+    let!(:pc) { create(:archive_copy, preserved_object: po, version: version) }
+    let(:zc) { pc.zip_checksums.first }
```

Comment:
That concern is effectively delegated to `#pcs`, not `#perform` (or `#update_size`).  I would be happy to follow up w/ a test of `#pcs` to that effect, if you like, but since it just relies on `PreservedCopy.for_archive_endpoints`, it seems like a test of that scope would cover it.